Assets/Scripts/Combat/ProjectileLaser.cs

```diff
@@ -15,11 +15,10 @@ public override void Start()
         _destroyOnHit = false;
         _dealDamageAfterSeconds = 0.5f;
 
-        if (_weapon == null) { GameAssetsManager.Instance.EnemyDamageDataSO.CreateDamageForEnemies(_damageTypes, _aiCombatID, ref _damageData); }
-        else 
+        _damageData = DamageData.GetDamageData(_damageTypes, _weapon, _aiCombatID);
+
+        if (_weapon != null)
```

Comment:
All the code inside of `if (_weapon != null)`  is repeated in `Projectile`, make `Projectile` handle this section of code